airbyte-integrations/connectors/source-hubspot/sample_files/configured_catalog.json

```diff
@@ -216,6 +130,12 @@
             "contacts": {
               "type": ["null", "array"]
             },
+            "createdAt": {
+              "type": "string"
+            },
+            "updatedAt": {
+              "type": "string"
+            },
```

Comment:
the `cursor_field` and `default_cursor_field` should be set on any stream that is using incremental. Is `discover` not outputing a default cursor field?

Reply:
but if `source_defined_cursor` is `True`, why `cursor_field` should be in catalog? I thought this one is coming from user, the same question about `default_cursor_field`, I thought this one used in case stream doesn't define `cursor_field` and user doesn't provide the `cursor_field`

Reply:
current implementation of the BaseClient return only JSON schema

Reply:
`cursor_field` is used by destinations when deduping records. For example, if the records `user=eugene,score=300,updatedAt=1` and `user=eugene,score=200,updatedAt=2` appear in the destination, assuming `user` is the primary key then we need to know what field to dedupe on i.e: cursor field

Reply:
This will also be used by standard tests to fix the incremental problem you are running into by verifying all records have cursor value >= state

Reply:
but these are two completely different use cases, I don't see how this will work in case there is no `primary_key` or `primary_key` is composite (multiple keys).
`update_at`/`bookmark_field` can't be used instead of `primary_key` in many cases because it means a different thing.

Reply:
I agree. `primary_key` is coming though :) see #2337